tap_outbrain/__init__.py

```diff
@@ -295,77 +296,56 @@ def sync_campaigns(state, access_token, account_id):
 
     LOGGER.info('Done!')
 
+def do_discover():
+    LOGGER.info("Starting discovery")
+    catalog = discover()
+    json.dump(catalog.to_dict(), sys.stdout, indent=2)
+    LOGGER.info("Finished discover")
 
 def do_sync(args):
     #pylint: disable=global-statement
     global DEFAULT_START_DATE
     state = DEFAULT_STATE
 
-    with open(args.config) as config_file:
-        config = json.load(config_file)
-        CONFIG.update(config)
+    config = args.config
+    CONFIG.update(config)
 
-    missing_keys = []
-    if 'username' not in config:
-        missing_keys.append('username')
-    else:
-        username = config['username']
-
-    if 'password' not in config:
-        missing_keys.append('password')
-    else:
-        password = config['password']
-
-    if 'account_id' not in config:
-        missing_keys.append('account_id')
-    else:
-        account_id = config['account_id']
-
-    if 'start_date' not in config:
-        missing_keys.append('start_date')
-    else:
-        # only want the date
-        DEFAULT_START_DATE = config['start_date'][:10]
-
-    if missing_keys:
-        LOGGER.fatal("Missing {}.".format(", ".join(missing_keys)))
-        raise RuntimeError
-
-    access_token = config.get('access_token')
-
-    if access_token is None:
-        access_token = generate_token(username, password)
+    DEFAULT_START_DATE = config.get('start_date')[:10]
 
+    access_token = config.get('access_token') or generate_token(config.get('username'), config.get('password'))
     if access_token is None:
         LOGGER.fatal("Failed to generate a new access token.")
         raise RuntimeError
 
     # NEVER RAISE THIS ABOVE DEBUG!
     LOGGER.debug('Using access token `{}`'.format(access_token))
 
+    with open("tap_outbrain/schemas/campaign.json") as f:
+        campaign = json.load(f)
+
+    with open("tap_outbrain/schemas/campaign_performance.json") as f:
+        campaign_performance = json.load(f)
 
     singer.write_schema('campaigns',
-                        schemas.campaign,
+                        campaign,
                         key_properties=["id"])
     singer.write_schema('campaign_performance',
-                        schemas.campaign_performance,
+                        campaign_performance,
                         key_properties=["campaignId", "fromDate"],
                         bookmark_properties=["fromDate"])
 
-    sync_campaigns(state, access_token, account_id)
+    sync_campaigns(state, access_token, config.get('account_id'))
 
 
 def main_impl():
-    parser = argparse.ArgumentParser()
-
-    parser.add_argument(
-        '-c', '--config', help='Config file', required=True)
-    parser.add_argument(
-        '-s', '--state', help='State file')
+    args = singer.utils.parse_args(
+        required_config_keys=['account_id', 'username', 'password', 'start_date']
+    )
```

Comment:
To stay with existing coding pattern, avoid using dangling paranthesis, something like below.

For readability, I would even prefer the define `required_config_keys` separately. 

```suggestion
    args = singer.utils.parse_args(required_config_keys=['account_id',
                                   'username',
                                   'password',
                                   'start_date'])
```

Reply:
Incorporated this.

Reply:
This is still not handled as expected.